spec/app/ics-030-middleware/README.md

```diff
@@ -94,6 +83,17 @@ interface ICS4Wrapper {
 }
 ```
 
+```typescript
+// AppWrapper is an interface designed to get information on the base IBC application
+// regardless of where you are in the current stack. This will allow middleware to act
+// on the base application data
+interface AppWrapper {
+    getAppVersion(portID: Identifier, channelId: Identifier, version: string): (string)
+    getAppPacketData(portID: Identifier, channelId: Identifier, packetData: bytes): (bytes)
+    getAppAcknowledgement(portID: Identifier, channelId: Identifier, acknowledgement: bytes): (bytes)
+}
```

Comment:
How do these functions work exactly? any app in the stack would call the app above it?